frontend/DataView/DataView.js

```diff
@@ -35,20 +35,51 @@ type DataViewProps = {
 class DataView extends React.Component {
   props: DataViewProps;
 
+  parseSparseArray(arr: Object) {
+    var acc: Array<any> = [];
+    var length = arr.length;
+    var numOfUndefsInRow = 0;
+    var pushUndefValsNumber = () => acc.push('undefined x ' + numOfUndefsInRow);
```

Comment:
Let's write this inline instead of an extra closure. Just duplicate this line if you need to.